File: 1.java
import java.util.Scanner;

public class MaxDigitSubstring {

    public static String findMaxSumSubstring(String s) {
        int maxSum = 0;
        StringBuilder result = new StringBuilder();
        int currentSum = 0;
        StringBuilder currentSub = new StringBuilder();
        boolean hasDigits = false;

        for (char c : s.toCharArray()) {
            if (Character.isDigit(c)) {
                hasDigits = true;
                currentSub.append(c);
                currentSum += Character.getNumericValue(c);
            } else {
                if (currentSub.length() > 0) {
                    if (currentSum > maxSum) {
                        maxSum = currentSum;
                        result = new StringBuilder(currentSub);
                    }
                    currentSub.setLength(0);
                    currentSum = 0;
                }
            }
        }

        // Проверяем последнюю подстроку
        if (currentSub.length() > 0 && currentSum > maxSum) {
            result = new StringBuilder(currentSub);
        }

        return hasDigits ? result.toString() : "В строке нет цифр!";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите строку: ");
        String input = scanner.nextLine();
        
        String result = findMaxSumSubstring(input);
        System.out.println("Результат: " + result);
        
        scanner.close();
    }
}

File: 1.py
def find_max_sum(s):
    max_sum = 0
    result = ""
    current_sum = 0
    current_sub = ""
    has_digits = False

    for char in s:
        if char.isdigit():
            has_digits = True
            current_sub += char
            current_sum += int(char)
        else:
            if current_sub:
                if current_sum > max_sum:
                    max_sum = current_sum
                    result = current_sub
                current_sub = ""
                current_sum = 0

    # Проверяем последнюю подстроку
    if current_sub:
        if current_sum > max_sum:
            result = current_sub

    return result if has_digits else "В строке нет цифр"
user_input = input("Введите строку: ")
print(find_max_sum(user_input))
